src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -27,8 +29,97 @@ interface Props {
+  setDownloadInProgress: (boolean) => void,
+  downloadStatusMessage: {
+    title: string,
+    message: string,
+    active: boolean,
+    url: string
+  },
+  setDownloadStatusMessage: (Object) => void;
 }
 
+const handleDownloadZipClick = async (
+  selectedResources: any[],
+  setDownloadInProgress: (boolean) => void,
+  setDownloadStatusMessage: (object) => void,
+) => {
+  const studyIDs = selectedResources.map((study) => study.project_number);
+  setDownloadInProgress(true);
+  setDownloadStatusMessage({
+    title: 'Your download is being prepared',
+    message: 'Please remain on this page while your download is being prepared.\n\n'
+               + 'When your download is ready, it will begin automatically. You can close this window.',
+    active: true,
+    url: '',
+  });
+
+  const handleJobError = (err) => {
+    console.error(err);
+    setDownloadInProgress(false);
+    setDownloadStatusMessage({
+      title: 'Download failed',
+      message: 'There was a problem preparing your download.'
+                + 'Please consider using the gen3 client to download these files via a manifest.',
+      active: true,
+      url: '',
+    });
+  };
+
+  fetchWithCreds({
+    path: `${jobAPIPath}dispatch`,
+    method: 'POST',
+    body: JSON.stringify({
+      action: 'batch-export',
+      input: {
+        study_ids: studyIDs,
+      },
+    }),
+  }).then(
+    (dispatchResponse) => {
+      const { uid } = dispatchResponse.data;
+      const pollForJobStatusUpdate = () => {
+        fetchWithCreds({ path: `${jobAPIPath}status?UID=${uid}` }).then(
+          (statusResponse) => {
+            const { status } = statusResponse.data;
+            if (status === 'Failed') {
+              fetchWithCreds({ path: `${jobAPIPath}output?UID=${uid}` }).then(
+                (outputResponse) => {
+                  setDownloadStatusMessage({
+                    title: 'Download failed',
+                    message: outputResponse.data.output,
+                    active: true,
+                    url: '',
+                  });
+                  setDownloadInProgress(false);
+                },
+              ).catch(handleJobError);
+            } else if (status === 'Completed') {
+              fetchWithCreds({ path: `${jobAPIPath}output?UID=${uid}` }).then(
+                (outputResponse) => {
+                  const { output } = outputResponse.data;
+                  setDownloadStatusMessage({
+                    title: 'Your download is ready',
+                    message: 'Your download has been prepared. If your download doesn\'t start automatically, please follow this direct link:',
+                    active: true,
+                    url: output,
+                  });
+                  setDownloadInProgress(false);
+                  setTimeout(() => window.open(output), 2000);
+                },
+              ).catch(handleJobError);
+            } else {
+              setTimeout(pollForJobStatusUpdate, 5000);
```

Comment:
This will cause the component stuck in an infinite loop if `uid` is not returned from https://github.com/uc-cdis/data-portal/pull/906/files#diff-1ac68dcc0067af6b35b142c058129318cc50ee20bfa9195afdb3d99f150d9d15R81

You can reproduce it now in qa-heal, since the mock auth is turned on, you will be login by default as a test account, which doesn't have `sower` access. The job dispatch request will return as 403 error, and no `uid` is returned, but this section of code will repeatly checking with `uid=undefined`

Better to handle fetched response with HTTP status code, rather than plain text

Reply:
True. I will add some checks for HTTP status code for cases like this. For the `/status` calls, however, the status code from Sower is 200 regardless of job state, so I will have to keep some plain text checks in there

Reply:
make sense! probably only the first call to `/job/dispatch` needs a status code check